Components/Core/Inc/Timer.hpp

```diff
@@ -25,21 +35,32 @@ constexpr uint32_t DEFAULT_TIMER_COMMAND_WAIT_PERIOD = MS_TO_TICKS(15); // Defau
 class Timer
 {
 public:
-	Timer();
-
-	bool ChangePeriod(const uint32_t period);
-
-	// WORK-IN-PROGRESS
-	// NOTES:
-	// - I can think of several timer types
-	// 1) Default Ctor Timer (1 second polling timer that requires polling to acquire state with no callback)
-	// 2) Callback Enabled Timer (user-provided callback)
+	Timer(); // Default Constructor (Polling Timer)
+	Timer(void (*TimerCallbackFunction_t)( TimerHandle_t xTimer )); // Constructor for Callback Enabled Timer
+	~Timer();
+	bool ChangePeriod(const uint32_t period_ms);
+	bool ChangePeriodAndStart(const uint32_t period_ms);
+	bool Start();
+	bool Stop();
+	bool ResetTimer();
+	bool ResetTimerAndStart();
+	void SetAutoReload(bool setReloadOn);
 
+	bool GetAutoReload();
+	TimerState GetState();
+	const uint32_t GetPeriod();
+	const uint32_t GetRemainingTime();
```

Comment:
Not a huge deal, but these getters could all return const and be const functions, eg:
```
const bool GetAutoReloadEnabled() const;
```

Would be nice to rename slightly ambiguous things like:
* GetAutoReload to GetAutoReloadEnabled
* GetPeriod to GetPeriodMs() -- or whatever unit it is
* GetRemainingTime to GetRemainingTimeMs() -- or whatever unit it is

Reply:
do you mind giving the names one more check i changed some

Reply:
yep, don't see changes though, maybe you haven't commit/push?